File: C++/level 3/N least common multiple.cpp
#include<iostream>
#include<vector>
using namespace std;
long long gcd(long long a, long long b){
	long long c;
	if(b == 0) 
    return a; 
  return gcd(b, a%b);
}

long long lcm(long long a,long long b){
    return a * b / gcd(a, b);
}

long long nlcm(vector<int> num){
	long long answer = 0;
  long long tem=0;
  for(int a=0;a<num.size(); a++){
    if(a==0)
    	tem = lcm(num[a],num[a+1]);
    else
      tem = lcm(num[a],tem);
  }
	return tem;
}

int main(){
	vector<int> test{2,11,45,54,56,62,64,73,82,84};

	// 아래는 테스트로 출력해 보기 위한 코드입니다.
	cout << nlcm(test);
}

File: Java/level 3/N least common multiple.java
class NLCM {
  public long gcd(long a, long b){
	if(b == 0) 
    return a; 
  return gcd(b, a%b);
	}

	public long lcm(long a,long b){
    return a * b / gcd(a, b);
	}
	public long nlcm(int[] num) {
	long answer = 0;
  long tem=0;
  for(int a=0;a<num.length; a++){
    if(a==0)
    	tem = lcm(num[a],num[a+1]);
    else
      tem = lcm(num[a],tem);
  }
	return tem;
	}
  
	public static void main(String[] args) {
		NLCM c = new NLCM();
		int[] ex = { 2, 6, 8, 14 };
		// 아래는 테스트로 출력해 보기 위한 코드입니다.
		System.out.println(c.nlcm(ex));
	}
}